assets/js/modules/analytics-4/utils/extract-dashboard-data.js

```diff
@@ -108,9 +108,7 @@ export function extractAnalytics4DashboardData(
 
 	// Pad rows to 2 x number of days data points to accommodate new accounts.
 	if ( days * 2 > rowLength ) {
-		// Cloning the reference date to be able to change it in the padding
-		// loop without affecting the actual reference date.
-		const date = new Date( referenceDate.getTime() ); // eslint-disable-line sitekit/no-direct-date
+		const date = new Date( referenceDate ); // eslint-disable-line sitekit/no-direct-date
```

Comment:
Please use `stringToDate` instead, otherwise it will return differently depending on the local timezone:

<img width="545" alt="image" src="https://github.com/user-attachments/assets/73203f0b-4c22-43af-a6e8-3535f8899dca">

See https://github.com/google/site-kit-wp/blob/62264e62fe3cd247b4f47f0676596bbc1c8d2e1d/assets/js/util/dates.js#L155-L163